PL_2_komentarze.py

```diff
@@ -0,0 +1,2722 @@
+        for col_index in range(0,3):
+            wsn1X.write(liczbawierszy, col_index, daneklienta[col_index])
+        wbn1X.save(adresklienta)
+        messagebox.showinfo("Dodano klienta do listy!", "Dodano klienta do listy!\n\nGratuluję sukcesu!")
+        ladujklientow()
+    else:
+        pass
+
+tree_columns = ("Nazwa klienta", "Adres 1", "Adres 2")
+def sortby(tree, col, descending):
+    """Sort tree contents when a column is clicked on."""
+    # grab values to sort
+    data = [(tree.set(child, col), child) for child in tree.get_children('')]
+
+    # reorder data
+    data.sort(reverse=descending)
+    for indx, item in enumerate(data):
+        tree.move(item[1], '', indx)
+
+    # switch the heading so that it will sort in the opposite direction
+    tree.heading(col,
+        command=lambda col=col: sortby(tree, col, int(not descending)))
+
+class Drzewo(object):
+    def __init__(self, rodzic, tree_data, entry1, entry2, entry3):
+        self.tree = None
+        self._setup_widgets(rodzic)
+        self._build_tree(tree_data)
+        self.tree.bind('<Double-1>', self.daneklienta)
+
+    def _setup_widgets(self, rodzic):
+        container = ttk.Frame(rodzic)
+#         container.pack(fill='both', expand=True)
+        container.grid()
+
+        # XXX Sounds like a good support class would be one for constructing
+        #     a treeview with scrollbars.
+        self.tree = ttk.Treeview(rodzic, columns=tree_columns, show="headings")
+        vsb = ttk.Scrollbar(rodzic, orient="vertical", command=self.tree.yview)
+        hsb = ttk.Scrollbar(rodzic, orient="horizontal", command=self.tree.xview)
+        self.tree.configure(yscrollcommand=vsb.set, xscrollcommand=hsb.set)
+        self.tree.grid(column=0, row=0, sticky='nsew')
+        vsb.grid(column=1, row=0, sticky='ns')
+        hsb.grid(column=0, row=1, sticky='ew')
+
+        container.grid_columnconfigure(0, weight=1)
+        container.grid_rowconfigure(0, weight=1)
+
+    def _build_tree(self, tree_data):
+        for col in tree_columns:
+            self.tree.heading(col, text=col.title(),
+                command=lambda c=col: sortby(self.tree, c, 0))
+            # XXX tkFont.Font().measure expected args are incorrect according
+            #     to the Tk docs
+            self.tree.column(col, width=font.Font().measure(col.title()))
+
+        for item in tree_data:
+            self.tree.insert('', 'end', values=item)
+
+            # adjust columns lenghts if necessary
+            for indx, val in enumerate(item):
+                ilen = font.Font().measure(val)
+                if self.tree.column(tree_columns[indx], width=None) < ilen:
+                    self.tree.column(tree_columns[indx], width=ilen)
+    
+    def daneklienta(self, event):
+        self.entryKlient1 = entryKlient1
+        self.entryKlient2 = entryKlient2
+        self.entryKlient3 = entryKlient3
+        item = self.tree.identify_row(event.y)
+        entryKlient1.delete(0, END)
```

Comment:
`entryKlient1..3` to dobre przykłady obiektów używanych w wielu miejscach w kodzie. Nie wygląda mi to na obiekt globalny, raczej na reużycie nazwy zmiennej w różnych miejscach. To jest spoko, ale tylko jeśli kod jest podzielony na klasy/moduły itp. Inaczej nie masz pewności co tak naprawdę znajdziesz w tym obiekcie albo używasz go globalnie i tworzy się nam spagetthi.